test/e2e/scalability/load.go

```diff
@@ -483,10 +486,16 @@ func GenerateConfigsForGroup(
 }
 
 func generateServicesForConfigs(configs []testutils.RunObjectConfig) []*v1.Service {
-	services := make([]*v1.Service, 0, len(configs))
+	services := make([]*v1.Service, 0)
+	currentSvcLabel := ""
 	for _, config := range configs {
+		svcLabel, found := config.GetLabelValue(svcLabelKey)
+		if !found || svcLabel == currentSvcLabel {
+			continue
+		}
+		currentSvcLabel = svcLabel
 		serviceName := config.GetName() + "-svc"
-		labels := map[string]string{"name": config.GetName()}
```

Comment:
Actually I just checked that this wasn't needed (though I can keep it if you want), as the Labels added are just extra. Unless some controller intentionally uses this service selector to select the pods, everything should be fine (and ideally this should happen only if the controller actually intends to modify pods of a service).